backend/src/personnel/personnel.service.ts

```diff
@@ -114,42 +114,48 @@ export class PersonnelService {
       });
     }
     /**
-     * If availabilityStatus is defined, check if availabilityStartDate and availabilityEndDate are defined - if not then, default to today's date, and return all peronnel with the availabilityStatus
+     * If we have an availability type, but no date range, we will default to the current date
      */
-    if (query.availabilityType) {
-      if (!query.availabilityFromDate || !query.availabilityToDate) {
-        qb.andWhere('availability.date =:date', {
-          date: format(new Date(), 'yyyy-MM-dd'),
-        });
-        qb.andWhere('availability.availabilityType = :availabilityType', {
-          availabilityType: query.availabilityType,
-        });
-      } else {
-        qb.andWhere('availability.availabilityType = :availabilityType', {
-          availabilityType: query.availabilityType,
-        });
-        qb.andWhere('availability.date BETWEEN :from AND :to', {
-          from: query.availabilityFromDate,
-          to: query.availabilityToDate,
-        });
-      }
+    if (
+      query.availabilityType &&
+      (!query.availabilityFromDate || !query.availabilityToDate)
+    ) {
+      qb.andWhere('availability.date =:date', {
+        date: format(new Date(), 'yyyy-MM-dd'),
+      });
+      qb.andWhere('availability.availabilityType = :availabilityType', {
+        availabilityType: query.availabilityType,
+      });
     }
     /**
-     * If availabilityStatus is not defined, check if availabilityStartDate and availabilityEndDate are defined - if not then, default to today's date, and return all peronnel with all availabilityStatus.
+     * If we have an availability type and a date range, we will use the date range + type
      */
-    if (!query.availabilityType) {
-      // This is the default view on pageload - all personnel and all status on today's date (if not indicated then the defualt status of not indicated is returned)
-      if (!query.availabilityFromDate || !query.availabilityToDate) {
-        qb.andWhere('availability.date =:date', {
-          date: format(new Date(), 'yyyy-MM-dd'),
-        });
-      } else {
-        // This query is not very meaningful without a status - returns all personnel with any status within the date range - we shoudl enforce a status to be selecred if searching by date range
-        qb.andWhere('availability.date BETWEEN :from AND :to', {
-          from: query.availabilityFromDate,
-          to: query.availabilityToDate,
-        });
-      }
+    if (
```

Comment:
I know these seem to be mutually exclusive, but I'd prefer to see else ifs here

Reply:
I agree. I'm hoping to remove the "if no status but dates" check by enforcing a status since the query results are somewhat meaningless without one on the front end. I have a TODO above the second if chunk to update if that is sufficient for the moment?

Reply:
Yeah sure